### PR TITLE
Added section about custom-route-generator

### PR DESCRIPTION
**Pull request by wachterjohannes**
This PR add the documentation for sulu PR https://github.com/sulu/sulu/pull/2820

### CODE REVIEW

File: bundles/route/custom_route_generator.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
+Custom Route-Generator
+======================
+
+The RouteBundle enables you to define custom Route-Generator.
```
Comment (danrot) [inline bundles/route/custom_route_generator.rst]: `route generators`

File: bundles/route/custom_route_generator.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+This Generator will be called everytime you create or update
```
Comment (danrot) [inline bundles/route/custom_route_generator.rst]: `These generators`

File: bundles/route/custom_route_generator.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+a route for any entity. For each entity you can define which
```
Comment (danrot) [inline bundles/route/custom_route_generator.rst]: `for the specified entity`? The point is that it is only called for one type of entities.

File: bundles/route/custom_route_generator.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+generator will be used.
+
+The following example will generate the route by using the
```
Comment (danrot) [inline bundles/route/custom_route_generator.rst]: `a route`

File: bundles/route/custom_route_generator.rst [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+core Route-Generator but you can define route-schemas for
```
Comment (danrot) [inline bundles/route/custom_route_generator.rst]: `route generator, but`

Comment (danrot) [inline bundles/route/custom_route_generator.rst]: `route schemas`

File: bundles/route/custom_route_generator.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+each type of an entity.
```
Comment (danrot) [inline bundles/route/custom_route_generator.rst]: `each entity type`

File: bundles/route/custom_route_generator.rst [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+.. code-block:: php
+
+    <?php
+
+    class RouteGenerator implements RouteGeneratorInterface
+    {
+        use TypeTrait;
```
Comment (danrot) [inline bundles/route/custom_route_generator.rst]: Can't find this trait in the codebase.

File: bundles/route/custom_route_generator.rst [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+         * {@inheritdoc}
+         */
+        public function getOptionsResolver(array $options)
+        {
+            // allow all options.
+            return (new OptionsResolver())->setDefined(array_keys($options));
+        }
+    }
+
+Register this as a service with the tag `sulu.route_generator` (with alias `by_type`
```
Comment (danrot) [inline bundles/route/custom_route_generator.rst]: Didn't we change the alias from `by_type` to something else?

File: bundles/route/custom_route_generator.rst [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+for example) and use this alias in the configuration:
+
+.. code-block:: yaml
+
+    sulu_route:
+        mappings:
+            AppBundle\Entity\Recipe:
+                generator: by_schema
```
Comment (danrot) [inline bundles/route/custom_route_generator.rst]: Same as above

File: bundles/route/custom_route_generator.rst [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+                options:
+                    route_schema: /{translator.trans('recipe')}/{object.getTitle()}
+            AppBundle\Entity\Example:
+                generator: by_type
```
Comment (danrot) [inline bundles/route/custom_route_generator.rst]: Same as above

File: bundles/route/custom_route_generator.rst (hunk 3 above)
Comment (wachterjohannes) [inline bundles/route/custom_route_generator.rst]: i think `for the given entity` is more specific.

**Comment by wachterjohannes** [comment]
@danrot fixed

File: bundles/route/index.rst [hunk 11]
```diff
@@ -64,7 +72,9 @@ Configure the route-schema in the file `app/config/config.yml`:
     sulu_route:
         mappings:
             AppBundle\Entity\Recipe:
-                route_schema: /{translator.trans('recipe')}/{object.getTitle()}
+                generator: by_schema
```
Comment (danrot) [inline bundles/route/index.rst]: Should also be `schema`, shouldn't it?

**Comment by wachterjohannes** [comment]
@danrot fixed

**Pull request merged**